client/src/components/__tests__/Signup.test.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { mount } from 'enzyme';
+
+import Signup from '../Signup';
+
+describe('<Signup/> form', () => {
+  const toggleModal = jest.fn();
+  const wrapper = mount(<Signup show toggleModal={toggleModal} />);
+
+  test('should return <FormDialog /> component', () => {
+    expect(wrapper.exists('FormDialog')).toBe(true);
+  });
+
+  test('should contain <form /> element', () => {
+    expect(wrapper.exists('form')).toBe(true);
+  });
+
+  test('form should have onsubmit method', () => {
+    expect(wrapper.find('form').props().onSubmit).toBeInstanceOf(Function);
+  });
+});
+
+describe('input fields in form', () => {
+  const toggleModal = jest.fn();
+  const wrapper = mount(<Signup show toggleModal={toggleModal} />);
+  const inputFields = wrapper.find('input');
+
+  test('should have 4 input fields', () => {
+    expect(inputFields.length).toBe(4);
+  });
+
+  test('should have email input field with onChange', () => {
```

Comment:
all four tests are same, create a util function then take an index, type and name as param and check the required expects

Reply:
Done.